activerecord/test/cases/relations_test.rb

```diff
@@ -370,6 +371,29 @@ def test_default_reverse_order_on_table_without_primary_key
     end
   end
 
+  def test_default_reverse_order_on_table_with_implicit_order_column
+    ordered_edge = Class.new(Edge) do
+      self.implicit_order_column = "source_id"
+    end
+
+    ordered_edge.delete_all
+
+    edge_1 = ordered_edge.create!(source_id: 1, sink_id: 2)
+    edge_2 = ordered_edge.create!(source_id: 2, sink_id: 3)
+    assert_equal edge_1.source_id, ordered_edge.all.first.source_id
+    assert_equal edge_2.source_id, ordered_edge.all.reverse_order.first.source_id
+  end
+
+  def test_default_reverse_order_on_table_with_query_constraints
+    ClothingItem.delete_all
+
+    ClothingItem.create!(clothing_type: "shirt")
+    ClothingItem.create!(clothing_type: "pant")
+    ClothingItem.create!(clothing_type: "tie")
+
+    assert_equal ["tie", "shirt", "pant"], ClothingItem.all.reverse_order.pluck(:clothing_type)
```

Comment:
```suggestion
    assert_equal ClothingItem.all.pluck(:clothing_type).reverse, ClothingItem.all.reverse_order.pluck(:clothing_type)
```
So that it doesn't break if someone adds a fixture.

Reply:
Addressed but I also sorted before reversing, cause the expectation query won't be ordered.